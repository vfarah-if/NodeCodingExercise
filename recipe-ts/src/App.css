.App {
  text-align: center;
  color: white;
}

.App-link {
  color: #61dafb;
}
