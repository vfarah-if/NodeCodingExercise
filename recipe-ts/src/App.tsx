import React from 'react';

import Header from './components/Header';
import MarsRoverImages from './components/MarsRoverImages';

import './App.css';
import GameOfLife from './components/GameOfLife';

function App() {
  let user = undefined;
  const loginHandler = () => alert('OnLogin ...');
  const createAccountHandler = () => alert('createAccountHandler ...');

  return (
    <div className='App'>
      <Header
        user={user}
        onLogin={loginHandler}
        onCreateAccount={createAccountHandler}
      />
      <br />
      <MarsRoverImages />
      <GameOfLife
        gridColor='green'
        activeColor='green'
        backgroundColor='red'
        boardSize={20}
        cellSize={50}
        seedActivePositions={[
          { x: 9, y: 4 },
          { x: 8, y: 5 },
          { x: 9, y: 5 },
          { x: 10, y: 5 },
          { x: 7, y: 6 },
          { x: 8, y: 6 },
          { x: 9, y: 6 },
          { x: 10, y: 6 },
          { x: 11, y: 6 },

          { x: 9, y: 15 },
          { x: 8, y: 14 },
          { x: 9, y: 14 },
          { x: 10, y: 14 },
          { x: 7, y: 13 },
          { x: 8, y: 13 },
          { x: 9, y: 13 },
          { x: 10, y: 13 },
          { x: 11, y: 13 },
        ]}
        showCellInfo={false}
      />
    </div>
  );
}

export default App;
