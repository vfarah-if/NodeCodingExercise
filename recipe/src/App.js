import './App.css';
import Button from './components/Button';
import Header from './components/Header';
import Hello from './components/Hello';
import UserDetails from './components/UserDetails';
import MarsRoverImages from './components/MarsRoverImages';

function App() {
	let user = undefined;
	const loginHandler = () => alert('OnLogin ...');
	const createAccountHandler = () => alert('createAccountHandler ...');
	const userId = 1;

	return (
		<div className="App">
			<Header
				user={user}
				onLogin={loginHandler}
				onCreateAccount={createAccountHandler}
			></Header>
			<header className="App-header">
				<MarsRoverImages />
				<p>
					Button Component to help understand setting up storyboarding
					from scratch
				</p>
				<Button
					backgroundColor="green"
					size="medium"
					displayName="Big Button"
					primary={true}
					onClick={() => alert('Hello from big button')}
					disabled={false}
				></Button>
				<Hello></Hello>
				<UserDetails id={userId}></UserDetails>				
			</header>
		</div>
	);
}

export default App;
