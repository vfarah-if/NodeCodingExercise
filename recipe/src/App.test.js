import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";

describe("given App", () => {
	test("should render hello stranger component", () => {
		render(<App />);

		const linkElement = screen.getByText(/Hello stranger/i);

		expect(linkElement).toBeInTheDocument();
	});

	test("should render acme header component", () => {
		render(<App />);

		const linkElement = screen.getByText(/Acme/i);

		expect(linkElement).toBeInTheDocument();
	});

	test("should render login button", () => {
		const alertSpy = jest.spyOn(window, 'alert');
		render(<App />);

		const linkElement = screen.getByText(/Log in/i);
		
		expect(linkElement).toBeInTheDocument();
		fireEvent.click(linkElement);
		expect(alertSpy).toHaveBeenCalledWith('OnLogin ...');
	});

	test("should render sign up button", () => {
		const alertSpy = jest.spyOn(window, 'alert');
		render(<App />);

		const linkElement = screen.getByText(/Sign up/i);
		
		expect(linkElement).toBeInTheDocument();
		fireEvent.click(linkElement);
		expect(alertSpy).toHaveBeenCalledWith('createAccountHandler ...');
	});

	test("should render big button component", () => {
		render(<App />);

		const linkElement = screen.getByText(/Big Button/i);
		const alertSpy = jest.spyOn(window, 'alert');

		expect(linkElement).toBeInTheDocument();
		fireEvent.click(linkElement);
        expect(alertSpy).toHaveBeenCalledWith('Hello from big button');

	});

	test("should render mars rover component", () => {
		render(<App />);

		const linkElement = screen.getByText(/Data provided by NASA/i);
	

		expect(linkElement).toBeInTheDocument();
	});

	test("should render a user details component", () => {
		const component = render(<App />);
		
		const linkElement = component.container.querySelector('li[class="card"]');

		expect(linkElement).toBeDefined();
		expect(linkElement).toMatchSnapshot();
	});

});
