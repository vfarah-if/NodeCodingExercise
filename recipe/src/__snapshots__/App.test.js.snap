// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given App should render a user details component 1`] = `
<li
  class="card"
>
  <p>
    No user details for User id '
    1
    '
  </p>
</li>
`;
